### PR TITLE
Add fyneselfupdate.NewConfig helper.

### PR DESCRIPTION
**Pull request by Bluebugs**

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import (
+	"crypto/ed25519"
+	"time"
+
+	"fyne.io/fyne/v2"
+	"github.com/fynelabs/selfupdate"
+)
+
+// NewConfig create a selfupdate.Config structure that provide Fyne UI dialog to progress the update
+func NewConfig(app fyne.App, win fyne.Window, source selfupdate.Source, schedule selfupdate.Schedule, publicKey ed25519.PublicKey) *selfupdate.Config {
```
Comment (andydotxyz) [inline config.go:12]: Aren't schedule and publicKey optional?
If so let's leave them off here, constructors should be mandatory really.

Comment (Bluebugs) [reply config.go:12]: PublicKey definitively not. I could make schedule optional, but it isn't at the moment. What would be the expected default behavior if there is no schedule set?

Comment (andydotxyz) [reply config.go:12]: Hmm, not sure about missing schedule - maybe just check on start?
Maybe it's not optional.
My concern about the key is that this API is now tied to `ed25519` - I think it should be `crypto.PublicKey`, that way this is much more flexible.

Comment (Bluebugs) [reply config.go:12]: Being tied to ed25519 is a opinionated choice here. I think if any alternate signature mechanism should be provided, they should be explicit and part of a new API or a new breaking version of the API.

**Review by andydotxyz — APPROVED** [review]

**Pull request merged**